### ISSUE TITLE
[DOCS]: Etsy API v3 Spec, shipping_profile_id doesn't fit into int

### ISSUE THREAD

**Issue by nh43de** [etsy/open-api#326]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Link to Documentation Issue

https://developers.etsy.com/documentation/reference#operation/createDraftListing

### What would you like to change?

shipping_profile_id has 12 digits and don't fit into int. This causes problems for automatically generated clients. The type for the OpenAPI spec should be int64 or uint64.

Source:
[https://developers.google.com/discovery/v1/type-format](https://developers.google.com/discovery/v1/type-format)
integer | int32 | A 32-bit signed integer. It has a minimum value of -2,147,483,648 and a maximum value of 2,147,483,647 (inclusive).

(note, same issue exists for Listing_image_id, it should be int64)

### Why are you proposing this change?

When sending or parsing shipping_profile_ids, this causes runtime errors for clients that are generated using the int type for shipping_profile_id.

**Comment by Gareth064** [comment]
If I recall rightly, the Open API spec doesn't have the concept of types of integers. So that's why it just says "integer" in the Etsy spec. 

I tried using a client generator for c# against the open API spec and this was one of the issues I came across too.

A rule of thumb I have taken with the Etsy API is whenever a property is an "**id**" then to make sure that it is defined as **long** in my code 

I had two options, either generate the client then start finding all ids and replacing them with long, and then fix everything that breaks. Or write my own generator.

I went with the second option. I have written, and will soon make public on my GitHub, a small c# console app which which generate all the models from the Etsy API spec specifically. In it I have written in the quirks with IDs needing be long and not int etc.

But the main takeaway is that there is nothing wrong with Etsy API spec, it is built against the proper Open API Spec which itself doesn't define integers differently.
